### PR TITLE
Validate  model

### PR DESCRIPTION
**Pull request by osamaashraf6**
## What does this PR do?

This PR is raised to implement the following features: 👍 :

> In this project, The Blog app will be a classic example of a blog website. I will create a fully functional website that will show the list of posts and empower readers to interact with them by adding comments and liking posts.🪄🎈

- [ ] Add validation For the User model: Name must not be blank.PostsCounter must be an integer greater than or equal to zero.
- [ ] Add validations to the Post model: The title must not be blank. The title must not exceed 250 characters. CommentsCounter must be an integer greater than or equal to zero. LikesCounter must be an integer greater than or equal to zero. .💯
- [ ] Open rails console for my project.
       Create and run the necessary migration files.👌

## Description of the task to be completed 👎 

- [ ] Run a program using the command line.👌
- [ ] Add validation For the User model: Name must not be blank.PostsCounter must be an integer greater than or equal to zero.
- [ ] Add validations to the Post model: The title must not be blank. The title must not exceed 250 characters. CommentsCounter must be an integer greater than or equal to zero. LikesCounter must be an integer greater than or equal to zero. .💯
- [ ] Initialize my project with Git.💯

  
## How should it be tested

Clone the repository
```
git clone https://github.com/osamaashraf6/Blog-App
```
Navigate to the folder
```
cd Blog-App
```

Checkout the branch
```
git checkout validate__model
```

#### Thanks for reviewing my PR 😊
Don't forget to give a ⭐️ to support me 😃

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
```
Comment (muneebulrehman) [inline spec/models/post_spec.rb:3]: - [ ] Kindly add a test for the `Title` where you will test the title must not exceed 250 characters.
- [ ] Kindly add a test for `Post` where the title is valid.
- [ ] Kindly add a test for the comments counter where it is valid
- [ ] Kindly add a test for the likes counter where it is valid

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
```
Comment (muneebulrehman) [inline spec/models/user_spec.rb:3]: - [ ] Kindly add a test for the name of the author, where it is valid.
- [ ] Kindly add a test for the post counter where it is valid

File: spec/models/like_spec.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe Like, type: :model do
```
Comment (muneebulrehman) [inline spec/models/like_spec.rb:3]: - [ ] Kindly test for the authorId where it is valid. I see you have created the user, post, and like. I would recommend testing whether the like belongs to the author who created it or not.  You have tested for its invalid part, now you can test for the valid part.

File: spec/models/comment_spec.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
```
Comment (muneebulrehman) [inline spec/models/comment_spec.rb:3]: - [ ] Kindly add a test for the post id to be valid.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @osamaashraf6 , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR
- All tests passing.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AnselemOdims — APPROVED** [review]
### Hi  @osamaashraf6 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**